docs/QuestionsIssuesPrs.md

```diff
@@ -20,95 +20,166 @@ If you don't get an answer within 1-2 days, feel free to post another comment in
+     > * You should not do more than one `d.FirstTimers` issues.<br>
+     > * You should wait till the first PR is merged before starting on more PRs. 
+     >   After you have one merged PR, you can work on multiple PRs in parallel.
+     >   This restriction aims to save you from making the same mistake in multiple PRs.
+     
+   * If the issue list does not contain what you want to work on, post an issue first (as described [above](#posting-an-issue))
+     and wait for it to be acknowledged. Otherwise you could end up fixing something that does not need fixing.
    * It is not required to (but OK to) ask for permission to work on an issue. 
-   But do check the issue discussion thread to see if there are _active_ PRs for that issue. 
-   If someone is already working on that issue, perhaps your efforts are better spent on a different issue.
+     But do check the issue discussion thread to see if there are _active_ PRs for that issue. 
+     If someone is already working on that issue, perhaps your efforts are better spent on a different issue.<br>
+     However, it is acceptable to submit PRs containing alternative solutions even if there are active PRs for an issue.
    * If it seems the existing PRs for that issue are no longer active (e.g. no activity in the past seven days), 
-   you can always post a message to check if anybody is still working on that issue.
-
-1. Create a branch with the name that follows the [naming conventions for branch names](FormatsAndConventions.md#branch).
-
-1. Fix the issue in the new branch. 
-   * Remember to adhere to coding style and testing requirements of the project, as specified in the project 
-     documentation.
-   * A PR should bring the code from one complete state to another. That means when you update code, relevant
-     code comments, tests, user docs, and developer docs should be updated too.
-   * Commit code at suitable points. Follow the [conventions for commit messages](FormatsAndConventions.md#commit).
-
-1. When you think you are done with fix, create a new PR. Here are the steps: 
-   1. **Update a PR with upstream changes**: There is a chance that while you were write code for the PR, 
-      the `master` branch of the _upstream repo_ (i.e. the original repo you forked from) has been updated with new code. 
-      In that case you need to update your fork and your clone with that new code so that the only difference between 
-      the code in your branch and the upstream `master` branch is the code you wrote.  <br>
-      
-      > See: [How to update a PR with upstream changes](HowToGuides.md#update-a-pr-with-upstream-changes)
+     you can always post a message to check if anybody is still working on that issue.
+
+1. **Create a new branch** 
+
+   Create a branch off the `master` branch. 
+   Follow [our naming conventions for branch names](FormatsAndConventions.md#branch).
+   
+   > Common mistake: Branch off from the currently active branch instead of the `master` branch.
```

Comment:
I wonder if we should phrase this as a gigantic `WARNING` instead, e.g.

```
WARNING: Make sure you branch off master rather than the currently active branch.
Otherwise, you may get commits from other branches inside your PR.
```